sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -452,14 +466,36 @@ object FunctionRegistry {
       }
     }
 
-    val clazz = tag.runtimeClass
+    (name, (expressionInfo[T](name), builder))
+  }
+
+  /**
+   * Creates a function lookup registry for cast aliases (SPARK-16730).
```

Comment:
NIT: `...function lookup registry...` should that ben`...function registry lookup entry ...` or something similar?